org.openhab.binding.zigbee/src/main/java/org/openhab/binding/zigbee/handler/ZigBeeThingHandler.java

```diff
@@ -457,6 +472,17 @@ public void run() {
         scheduler.schedule(commandHandler, 0, TimeUnit.MILLISECONDS);
     }
 
+    @Override
+    public @Nullable StateDescription getStateDescription(Channel channel,
+            @Nullable StateDescription originalStateDescription, @Nullable Locale locale) {
+        if (stateDescriptions.containsKey(channel.getUID())) {
+            logger.trace("Returning new stateDescription for {}", channel.getUID());
+            return stateDescriptions.get(channel.getUID());
+        } else {
+            return originalStateDescription;
```

Comment:
From the JavaDoc of the `DynamicStateDescriptionProvider`, I take that this method should return `null` (and not the `originalStateDescription`) if no dynamic state description shall be provided. (Citation from JavaDoc: `@return state description or null if none provided`)

```suggestion
            return null;
```